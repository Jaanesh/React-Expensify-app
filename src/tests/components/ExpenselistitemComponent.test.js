import React from 'react';
import {shallow} from 'enzyme';
import moment from 'moment';
import ExpenseItem from '../../components/ExpenseListItemComponent';

let expenseObj1={id:'1', description:'jan rent',note:'',amount:3,createdAt:moment(0).
                                                                           subtract(5,'days').valueOf()};

test('should render a single expenseitem when one expense is passes as prop ',()=>{
     const wrapper=shallow(<ExpenseItem {...expenseObj1}/>);
     expect(wrapper).toMatchSnapshot();
});

test('should render a  expenseitem when no expense is passes as prop ',()=>{
    const wrapper=shallow(<ExpenseItem />);
    expect(wrapper).toMatchSnapshot();
});

