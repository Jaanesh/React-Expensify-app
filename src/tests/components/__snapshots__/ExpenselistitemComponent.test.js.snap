// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a  expenseitem when no expense is passes as prop  1`] = `
<tr>
  <td>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/edit/undefined"
    />
  </td>
  <td />
  <td />
</tr>
`;

exports[`should render a single expenseitem when one expense is passes as prop  1`] = `
<tr>
  <td>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/edit/1"
    >
      jan rent
    </NavLink>
  </td>
  <td>
    3
  </td>
  <td>
    -432000000
  </td>
</tr>
`;
